assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -54,13 +90,108 @@ BlockDraggable.propTypes = {
+		const { setCurrentPage } = dispatch( 'amp/story' );
+		const { selectBlock, removeBlock, insertBlock, updateBlockAttributes } = dispatch( 'core/block-editor' );
+
+		const isCTABlock = 'amp/amp-story-cta' === blockName;
+
+		/**
+		 * Is the current element allowed on a given page?
+		 *
+		 * This is currently only used to filter out instances, where CTA's aren't allowed.
+		 *
+		 * @param {number} targetPageIndex  Zero-based index of new page
+		 * @param {string} targetPageId  Block id of new page
+		 * @return {boolean} Returns true if this element is allowed on the given page, false otherwise.
+		 */
+		const isElementAllowedOnPage = ( targetPageIndex, targetPageId ) => {
+			if ( isCTABlock ) {
+				// CTA's aren't allowed on the first page
+				if ( targetPageIndex === 0 ) {
+					return false;
+				}
+
+				// CTA's aren't allowed on pages, that already have a CTA
+				const CTABlockOnTargetPage = getCallToActionBlock( targetPageId );
+				const hasCTABlockOnTargetPage = Boolean( CTABlockOnTargetPage );
+				if ( hasCTABlockOnTargetPage ) {
+					return false;
+				}
+			}
+
+			return true;
+		};
+
+		/**
+		 * Get id of neighbor page that is `offset` away from the current page.
+		 *
+		 * If no page exists in that direction, if offset is zero or if element is not allowed on that page, null will be returned.
+		 *
+		 * @param {number} offset  Integer specifying offset from current page - e.g. -2 on page 3 will return id of page 1. Page indices are zero-based.
+		 * @return {string} Returns id of target page or null if element can't be dropped there for any reason.
+		 */
+		const getNeighborPageId = ( offset ) => {
+			const pages = getBlockOrder();
+			const currentPageIndex = pages.findIndex( ( i ) => i === rootClientId );
+			const newPageIndex = currentPageIndex + offset;
+			const isInsidePageCount = newPageIndex >= 0 && newPageIndex < pages.length;
+			const newPageId = pages[ newPageIndex ];
+			const isAllowedOnPage = isElementAllowedOnPage( newPageIndex, newPageId );
+
+			// Do we even have a legal neighbor in that direction? (offset=0 is not a neighbor)
+			if ( offset === 0 || ! isInsidePageCount || ! isAllowedOnPage ) {
+				return null;
+			}
+
+			return newPageId;
+		};
+
+		/**
+		 * Drop this element on the page `offset` away from the current page (if possible). If dropped, update the position of the element on the new page.
+		 *
+		 * Currently this function removes the old element and creates a clone on the new page.
+		 *
+		 * @param {number} offset  Integer specifying offset from current page - e.g. -2 on page 3 will return id of page 1. Page indices are zero-based.
+		 * @param {Object} newAttributes  Object with attributes to update on element on new page.
+		 */
+		const onNeighborDrop = ( offset, newAttributes ) => {
+			const newPageId = getNeighborPageId( offset );
+			if ( ! newPageId ) {
+				return;
+			}
+
+			// Remove block and add cloned block to new page.
+			removeBlock( clientId );
```

Comment:
Would be safer to remove the block last thing here?

Reply:
What could happen if done in reverse?

Reply:
This code reminds of code used for cutting blocks. There was an issue that was removed block before copied block was ready. This might not be an issue here, just came to mind when reviewing code. 

Wonder if this could / should reuse code from cut and paste? They basically do the same thing..

Reply:
The function internally use the `select`'ed function `getCallToActionBlock`, but that could of course be supplied to the helper. I'll look into it. But yes, it is a good candidate for unit test.

I don't really see an easy (or even a hard) combination with cut and paste. Functionality is overlapping, but far from similar.